q2cli/cli.py

```diff
@@ -0,0 +1,104 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016--, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import click
+import qiime
+from qiime.sdk import PluginManager, SubprocessExecutor
+
+from . import __version__ as q2cli_version
+
+@click.group()
+@click.pass_context
+def cli(ctx):
+    ctx.obj['plugin_manager'] = PluginManager()
+
+@cli.command()
+def version():
+    click.echo('QIIME version: %s' % qiime.__version__)
+    click.echo('q2cli version: %s' % q2cli_version)
+
+@cli.command()
+@click.pass_context
+def plugins(ctx):
+    installed_plugins = ctx.obj['plugin_manager'].plugins.items()
+    if len(installed_plugins) == 0:
+        click.echo('No plugins are currently installed.\nYou can browse the '
+                   'official QIIME 2 plugins at: https://github.com/qiime2.')
+    else:
+        click.echo('Installed plugins:')
+        for name, plugin in installed_plugins:
+            click.echo(' %s %s (%s)' %
+                       (name, plugin.version, plugin.website))
+
+# should the QIIME types be hashable to support this without casting
+# to a string?
+_type_map = {'Int': click.INT, 'Str': click.STRING, 'Float': click.FLOAT}
+
+def build_commands(group):
+    plugin_manager = PluginManager()
+    for plugin_name, plugin in plugin_manager.plugins.items():
+        for workflow_name, workflow in plugin.workflows.items():
+            parameters = []
+            for ia_name, ia_type in workflow.signature.input_artifacts.items():
```

Comment:
Just [followed up](https://github.com/qiime2/q2cli/issues/2#issuecomment-212004236) on #2, we should be able to support lazy loading of subcommands. Not blocking for this PR